### ISSUE TITLE
[Issue] Console Spamming/Duplication of items

### ISSUE THREAD

**Issue by GamingWubba93** [SeanOMik/EnergeticStorage#1]
**Describe the bug**
Was pulling out items and experienced a massive console spam and items were duplicating. Ended up removing the plugin from the server for the time being. 

**To Reproduce**
Place items like signs or cobblestone walls into a system with 5 64k drives. Shift click them to move them back into your inventory. You can even right click them to pull them into your inventory. 

**Screenshots / Video / Console Log**
[2020-06-24-3.log.gz](https://github.com/SeanOMik/EnergeticStorage/files/4822994/2020-06-24-3.log.gz)

**Server Information (please complete the following information):**
 - Minecraft Version Spigot 1.15.2
- Plugin Version 0.1-SNAPSHOT

**Comment by SeanOMik** [comment]
Okay, I'll take a look at it.

**Comment by SeanOMik** [comment]
It looks like its working fine for me. Could you send a list of plugins?

**Comment by GamingWubba93** [comment]
Yeah no problem! Sorry for getting back to you so late. 

List of Plugins:
![2020-07-01_23 07 27](https://user-images.githubusercontent.com/28677826/86312096-e7472a00-bbef-11ea-87ed-eae52221b5ba.png)

**Comment by SeanOMik** [comment]
Hm, could you try reinstalling the plugin and take a screenshot of the items that you put in the system as the errors occur? Maybe it has something to do with the order of items, or how you put the drives into the system. Also you could try to shift click the drives into the system as it may be an issue with that.

**Comment by GamingWubba93** [comment]
> Hm, could you try reinstalling the plugin and take a screenshot of the items that you put in the system as the errors occur? Maybe it has something to do with the order of items, or how you put the drives into the system. Also you could try to shift click the drives into the system as it may be an issue with that.

Alright I'll give that a go. Might it also be a good idea if I allow you to join the server as well?

**Comment by SeanOMik** [comment]
Yeah it may be, but I still wouldn't really be able to debug. Maybe if you wanted to talk via discord you could just send me a copy of your entire server so I could debug with an actual debugger. If you don't want to give me the entire server that's fine.

**Comment by GamingWubba93** [comment]
Lol tried to give myself some 64k drives. Ended up getting kicked with this message. 

> [20:49:30] [Server thread/INFO]: GamingWubba93 lost connection: Illegal characters in chat
> [20:49:30] [Server thread/INFO]: GamingWubba93 left the game

**Comment by SeanOMik** [comment]
Uhh, I don't think that's something caused by me. Did it show the message that you sent? And did it happen multiple times when you tried to give them to yourself?

**Comment by GamingWubba93** [comment]
It did not. It's possible it was because I tabbed my username into it. Yup That's what it was. 
![2020-07-04_16 55 53](https://user-images.githubusercontent.com/28677826/86520871-4d61c600-be17-11ea-92a1-957a7111f793.png)

So I haven't noticed the duplication glitch yet. Maybe mine was just a rare occurrence. I'll let you know if anything funky pops up! Feel free to close this issue.

**Comment by GamingWubba93** [comment]
Ahhh wait! If you pull out a drive from a system to move it to another system, thats when the duplication issue happens. 

> > [20:58:07] [Server thread/ERROR]: Could not pass event InventoryClickEvent to EnergeticStorage v0.1-SNAPSHOT
> org.bukkit.event.EventException: null
> at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:319) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:589) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:576) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.PlayerConnection.a(PlayerConnection.java:2148) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.PacketPlayInWindowClick.a(SourceFile:32) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.PacketPlayInWindowClick.a(SourceFile:10) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.PlayerConnectionUtils.lambda$0(PlayerConnectionUtils.java:19) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.TickTask.run(SourceFile:18) [spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.IAsyncTaskHandler.executeTask(SourceFile:144) [spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.IAsyncTaskHandlerReentrant.executeTask(SourceFile:23) [spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.IAsyncTaskHandler.executeNext(SourceFile:118) [spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.MinecraftServer.ba(MinecraftServer.java:918) [spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.MinecraftServer.executeNext(MinecraftServer.java:911) [spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.IAsyncTaskHandler.awaitTasks(SourceFile:127) [spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.MinecraftServer.sleepForTick(MinecraftServer.java:895) [spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.MinecraftServer.run(MinecraftServer.java:828) [spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at java.lang.Thread.run(Thread.java:748) [?:1.8.0_242]
> Caused by: java.lang.NullPointerException
> at net.seanomik.energeticstorage.gui.ESDriveGUI.initializeItems(ESDriveGUI.java:58) ~[?:?]
> at net.seanomik.energeticstorage.gui.ESDriveGUI.openInventory(ESDriveGUI.java:96) ~[?:?]
> at net.seanomik.energeticstorage.gui.ESSystemGUI.onInventoryClick(ESSystemGUI.java:207) ~[?:?]
> at sun.reflect.GeneratedMethodAccessor231.invoke(Unknown Source) ~[?:?]
> at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_242]
> at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_242]
> at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:315) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> ... 17 more
> > [20:58:09] [Server thread/ERROR]: Could not pass event InventoryCloseEvent to EnergeticStorage v0.1-SNAPSHOT
> org.bukkit.event.EventException: null
> at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:319) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:589) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:576) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at org.bukkit.craftbukkit.v1_15_R1.event.CraftEventFactory.handleInventoryCloseEvent(CraftEventFactory.java:1259) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.PlayerConnection.a(PlayerConnection.java:1927) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.PacketPlayInCloseWindow.a(PacketPlayInCloseWindow.java:18) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.PacketPlayInCloseWindow.a(PacketPlayInCloseWindow.java:1) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.PlayerConnectionUtils.lambda$0(PlayerConnectionUtils.java:19) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.TickTask.run(SourceFile:18) [spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.IAsyncTaskHandler.executeTask(SourceFile:144) [spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.IAsyncTaskHandlerReentrant.executeTask(SourceFile:23) [spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.IAsyncTaskHandler.executeNext(SourceFile:118) [spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.MinecraftServer.ba(MinecraftServer.java:918) [spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.MinecraftServer.executeNext(MinecraftServer.java:911) [spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.IAsyncTaskHandler.awaitTasks(SourceFile:127) [spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.MinecraftServer.sleepForTick(MinecraftServer.java:895) [spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.MinecraftServer.run(MinecraftServer.java:828) [spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at java.lang.Thread.run(Thread.java:748) [?:1.8.0_242]
> Caused by: java.lang.NullPointerException
> at net.seanomik.energeticstorage.files.PlayersFile.savePlayerSystem(PlayersFile.java:159) ~[?:?]
> at net.seanomik.energeticstorage.gui.ESDriveGUI.onInventoryClose(ESDriveGUI.java:152) ~[?:?]
> at sun.reflect.GeneratedMethodAccessor186.invoke(Unknown Source) ~[?:?]
> at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_242]
> at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_242]
> at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:315) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> ... 18 more
> > [20:58:13] [Server thread/ERROR]: Could not pass event InventoryClickEvent to EnergeticStorage v0.1-SNAPSHOT
> org.bukkit.event.EventException: null
> at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:319) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:589) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:576) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.PlayerConnection.a(PlayerConnection.java:2148) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.PacketPlayInWindowClick.a(SourceFile:32) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.PacketPlayInWindowClick.a(SourceFile:10) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.PlayerConnectionUtils.lambda$0(PlayerConnectionUtils.java:19) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.TickTask.run(SourceFile:18) [spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.IAsyncTaskHandler.executeTask(SourceFile:144) [spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.IAsyncTaskHandlerReentrant.executeTask(SourceFile:23) [spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.IAsyncTaskHandler.executeNext(SourceFile:118) [spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.MinecraftServer.ba(MinecraftServer.java:918) [spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.MinecraftServer.executeNext(MinecraftServer.java:911) [spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.IAsyncTaskHandler.awaitTasks(SourceFile:127) [spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.MinecraftServer.sleepForTick(MinecraftServer.java:895) [spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.MinecraftServer.run(MinecraftServer.java:828) [spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at java.lang.Thread.run(Thread.java:748) [?:1.8.0_242]
> Caused by: java.lang.IndexOutOfBoundsException: Index: 3, Size: 2
> at java.util.ArrayList.rangeCheck(ArrayList.java:657) ~[?:1.8.0_242]
> at java.util.ArrayList.remove(ArrayList.java:496) ~[?:1.8.0_242]
> at net.seanomik.energeticstorage.gui.ESDriveGUI.onInventoryClick(ESDriveGUI.java:226) ~[?:?]
> at sun.reflect.GeneratedMethodAccessor232.invoke(Unknown Source) ~[?:?]
> at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_242]
> at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_242]
> at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:315) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> ... 17 more
> > [20:58:13] [Server thread/ERROR]: Could not pass event InventoryClickEvent to EnergeticStorage v0.1-SNAPSHOT
> org.bukkit.event.EventException: null
> at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:319) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:589) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:576) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.PlayerConnection.a(PlayerConnection.java:2148) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.PacketPlayInWindowClick.a(SourceFile:32) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.PacketPlayInWindowClick.a(SourceFile:10) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.PlayerConnectionUtils.lambda$0(PlayerConnectionUtils.java:19) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.TickTask.run(SourceFile:18) [spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.IAsyncTaskHandler.executeTask(SourceFile:144) [spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.IAsyncTaskHandlerReentrant.executeTask(SourceFile:23) [spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.IAsyncTaskHandler.executeNext(SourceFile:118) [spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.MinecraftServer.ba(MinecraftServer.java:918) [spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.MinecraftServer.executeNext(MinecraftServer.java:911) [spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.IAsyncTaskHandler.executeAll(SourceFile:103) [spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.MinecraftServer.sleepForTick(MinecraftServer.java:894) [spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.MinecraftServer.run(MinecraftServer.java:828) [spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at java.lang.Thread.run(Thread.java:748) [?:1.8.0_242]
> Caused by: java.lang.IndexOutOfBoundsException: Index: 4, Size: 2
> at java.util.ArrayList.rangeCheck(ArrayList.java:657) ~[?:1.8.0_242]
> at java.util.ArrayList.remove(ArrayList.java:496) ~[?:1.8.0_242]
> at net.seanomik.energeticstorage.gui.ESDriveGUI.onInventoryClick(ESDriveGUI.java:226) ~[?:?]
> at sun.reflect.GeneratedMethodAccessor232.invoke(Unknown Source) ~[?:?]
> at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_242]
> at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_242]
> at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:315) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> ... 17 more
> > [20:59:00] [Craft Scheduler Thread - 6/INFO]: [Slimefun] Auto-Saving Block Data... (Next Auto-Save: 10m)
> > [20:59:00] [Craft Scheduler Thread - 6/INFO]: [Slimefun] Saving Blocks for World "world" (41,993 Change(s) queued)
> > [20:59:04] [Server thread/ERROR]: Could not pass event InventoryClickEvent to EnergeticStorage v0.1-SNAPSHOT
> org.bukkit.event.EventException: null
> at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:319) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:589) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:576) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.PlayerConnection.a(PlayerConnection.java:2148) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.PacketPlayInWindowClick.a(SourceFile:32) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.PacketPlayInWindowClick.a(SourceFile:10) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.PlayerConnectionUtils.lambda$0(PlayerConnectionUtils.java:19) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.TickTask.run(SourceFile:18) [spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.IAsyncTaskHandler.executeTask(SourceFile:144) [spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.IAsyncTaskHandlerReentrant.executeTask(SourceFile:23) [spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.IAsyncTaskHandler.executeNext(SourceFile:118) [spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.MinecraftServer.ba(MinecraftServer.java:918) [spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.MinecraftServer.executeNext(MinecraftServer.java:911) [spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.IAsyncTaskHandler.awaitTasks(SourceFile:127) [spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.MinecraftServer.sleepForTick(MinecraftServer.java:895) [spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.MinecraftServer.run(MinecraftServer.java:828) [spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at java.lang.Thread.run(Thread.java:748) [?:1.8.0_242]
> Caused by: java.lang.NullPointerException
> at net.seanomik.energeticstorage.gui.ESTerminalGUI.initializeItems(ESTerminalGUI.java:86) ~[?:?]
> at net.seanomik.energeticstorage.gui.ESTerminalGUI.openInventory(ESTerminalGUI.java:55) ~[?:?]
> at net.seanomik.energeticstorage.gui.ESSystemGUI.onInventoryClick(ESSystemGUI.java:203) ~[?:?]
> at sun.reflect.GeneratedMethodAccessor231.invoke(Unknown Source) ~[?:?]
> at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_242]
> at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_242]
> at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:315) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> ... 17 more
> > [20:59:08] [Server thread/ERROR]: Could not pass event InventoryCloseEvent to EnergeticStorage v0.1-SNAPSHOT
> org.bukkit.event.EventException: null
> at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:319) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:589) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:576) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at org.bukkit.craftbukkit.v1_15_R1.event.CraftEventFactory.handleInventoryCloseEvent(CraftEventFactory.java:1259) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.PlayerConnection.a(PlayerConnection.java:1927) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.PacketPlayInCloseWindow.a(PacketPlayInCloseWindow.java:18) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.PacketPlayInCloseWindow.a(PacketPlayInCloseWindow.java:1) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.PlayerConnectionUtils.lambda$0(PlayerConnectionUtils.java:19) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.TickTask.run(SourceFile:18) [spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.IAsyncTaskHandler.executeTask(SourceFile:144) [spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.IAsyncTaskHandlerReentrant.executeTask(SourceFile:23) [spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.IAsyncTaskHandler.executeNext(SourceFile:118) [spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.MinecraftServer.ba(MinecraftServer.java:918) [spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.MinecraftServer.executeNext(MinecraftServer.java:911) [spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.IAsyncTaskHandler.awaitTasks(SourceFile:127) [spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.MinecraftServer.sleepForTick(MinecraftServer.java:895) [spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at net.minecraft.server.v1_15_R1.MinecraftServer.run(MinecraftServer.java:828) [spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> at java.lang.Thread.run(Thread.java:748) [?:1.8.0_242]
> Caused by: java.lang.NullPointerException
> at net.seanomik.energeticstorage.files.PlayersFile.savePlayerSystem(PlayersFile.java:159) ~[?:?]
> at net.seanomik.energeticstorage.gui.ESTerminalGUI.onInventoryClose(ESTerminalGUI.java:204) ~[?:?]
> at sun.reflect.GeneratedMethodAccessor184.invoke(Unknown Source) ~[?:?]
> at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_242]
> at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_242]
> at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:315) ~[spigot-1.15.2.jar:git-Spigot-a99063f-8ea9b13]
> ... 18 more

**Comment by GamingWubba93** [comment]
If you'd like to join the server to experience it for yourself.


**149.56.28.188 | 2206**

**Comment by GamingWubba93** [comment]
> Ahhh wait! If you pull out a drive from a system to move it to another system, thats when the duplication issue happens.

Don't know if this helps, but after this happens and the ES System is broken it becomes a Hacked Computer.

**Comment by SeanOMik** [comment]
Okay now I have something I can test for myself. free I get off work I'll try to move a drive from a system. If I get any errors, I'll tell you.

**Comment by SeanOMik** [comment]
Okay well I tried to put some items in a system, took the drive out and then moved it to another system and everything worked as expected. I send a friend request to you in discord, same username as I have here.

**Comment by SeanOMik** [comment]
This was happening due to a custom implementation of hashCode in `ItemStack`. When the player removes items from the system I ran this code:
`itemStacks.entrySet().removeIf(entry -> removeAmountFromLore(entry.getKey()).isSimilar(item)); `
Where `itemStacks` is all items inside of the system and `item` is the item they're removing from the system. Java's implementation of `Iterator#remove` works by recalculating the `hashCode` so thats the reason this is not working. I fixed it by using some ugly code but it works. So until Spigot fixes this, I will have to stick with what I'm doing.
